go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -588,6 +588,23 @@ func (tsv *TabletServer) getPriorityFromOptions(options *querypb.ExecuteOptions)
 	return optionsPriority
 }
 
+// resolveTargetType returns the appropriate target tablet type for a
+// TabletServer request. If the caller has a local context then it's
+// an internal request and the target is the local tablet's current
+// target.
+func (tsv *TabletServer) resolveTargetType(ctx context.Context, target *querypb.Target) (string, error) {
+	if target != nil {
+		return target.TabletType.String(), nil
+	}
+	if !tabletenv.IsLocalContext(ctx) {
+		return topodatapb.ShardReplicationError_UNKNOWN.String(), vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "no target specified")
+	}
+	if tsv.sm.Target() == nil {
+		return topodatapb.ShardReplicationError_UNKNOWN.String(), vterrors.Errorf(vtrpcpb.Code_FAILED_PRECONDITION, "TabletServer has no current target")
+	}
+	return tsv.sm.Target().String(), nil
```

Comment:
This needs to be `tsv.sm.Target().TabletType` further converted to a string.
Interestingly:
https://github.com/vitessio/vitess/blob/main/go/vt/vttablet/tabletserver/tabletserver.go#L168-L172

Reply:
Looks like this has already been fixed in a later commit.

Reply:
Should we add a nil check on `sm.Target()` in line 172 while we are at it?